conduit/pipeline.go

```diff
@@ -339,12 +369,13 @@ func (p *pipelineImpl) Start() {
 					}
 					importTime := time.Since(start)
 					// Ignore round 0 (which is empty).
-					if p.round > 0 {
+					if p.pipelineMetadata.NextRound > 0 {
 						p.addMetrics(blkData, importTime)
 					}
 					// Increment Round
 					p.setError(nil)
-					p.round++
+					p.pipelineMetadata.NextRound++
+					p.encodeMetadataToFile()
```

Comment:
This is potentially a huge bottleneck on how quickly we can fully catchup from 0 on a new Indexer node. Not only are we doing blocking file IO between each round, but we are also creating a temp file and renaming it.

An idea: Maybe send completed rounds to a channel which is read by a goroutine that asynchronously keeps the metadata file up to date. We would need to ensure that the channel is fully flushed before properly exiting Conduit otherwise DB and metadata could be unsynched. If the process does not terminate gracefully of course they will have that problem anyways. It's potentially a bit more complex, but it seems to me like we might want that complexity here to get these operations off the main run loop.

Reply:
I'm leaving this unchanged for now. I ran a 15min block generator test locally and the average import time is no worse than one in conduit branch. I will set up an ec2 to run a longer test.